src/types/any.rs

```diff
@@ -2193,7 +2173,7 @@ impl<'py> PyAnyMethods<'py> for Bound<'py, PyAny> {
 
     #[cfg(not(PyPy))]
     fn py_super(&self) -> PyResult<Bound<'py, PySuper>> {
-        PySuper::new2(Bound::borrowed_from_gil_ref(&self.get_type()), self)
+        PySuper::new2(self.get_type().as_bound(), self)
```

Comment:
`new_bound` instead of `new2`?

Reply:
How would you feel if I just changed `PySuper::new` to return `Bound<T>` and same for `PyIterator::from_object`? I think that churn is probably in the category of acceptable. (And I was thinking we would do this for `PyTuple::new` and constructors in general.)

I guess we could also introduce these as `_bound` variants for 0.21 and deprecate the original names, and then change the return types of the original names later in 0.22 or 0.23?

Reply:
Actually now I think a little more it's probably indeed better to add `_bound` variants for 0.21 and just deprecate the old names? I'd rather break as little user code as possible given the size of the API change already.

Reply:
> Actually now I think a little more it's probably indeed better to add _bound variants for 0.21 and just deprecate the old names? I'd rather break as little user code as possible given the size of the API change already.

Yes, I think I would prefer that too. I guess there will eventually be a great reckoning release where we remove all the deprecated functionality and start transitioning to use the shorter names for the long-term approach.